langchain/src/text_splitter.ts

```diff
@@ -96,23 +96,39 @@ export abstract class TextSplitter
         let pageContent = chunkHeader;
 
         // we need to count the \n that are in the text before getting removed by the splitting
-        let numberOfIntermediateNewLines = 0;
+        const indexChunk = text.indexOf(chunk);
         if (prevChunk) {
-          const indexChunk = text.indexOf(chunk);
           const indexEndPrevChunk =
             text.indexOf(prevChunk) + (await this.lengthFunction(prevChunk));
-          const removedNewlinesFromSplittingText = text.slice(
-            indexEndPrevChunk,
-            indexChunk
-          );
-          numberOfIntermediateNewLines = (
-            removedNewlinesFromSplittingText.match(/\n/g) || []
-          ).length;
+          if (indexEndPrevChunk < indexChunk) {
+            const removedNewlinesFromSplittingText = text.slice(
```

Comment:
I know you didn't name this, but can you change this var name to `intermediateText` or something similar?